src/main/java/core/basesyntax/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.mapvalidator.KeyAbsentValidator;
+import java.util.Map;
+
+public class BalanceOperationHandler
+        extends AbstractOperationHandler
+        implements KeyAbsentValidator {
+    public BalanceOperationHandler() {
+        super(Operation.BALANCE);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                      FruitTransaction transaction) {
+        validateMap(fruitQuantityMap, transaction);
```

Comment:
Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. Here, you are using `validateMap` directly, which seems to be a part of a validation strategy. Consider refactoring this to use the Strategy pattern if applicable.